tests/integration/test_async_search_index.py

```diff
@@ -313,10 +312,87 @@ async def test_check_index_exists_before_search(async_client, async_index):
 
 
 @pytest.mark.asyncio
-async def test_check_index_exists_before_info(async_client, async_index):
-    await async_index.set_client(async_client)
+async def test_check_index_exists_before_info(async_index):
     await async_index.create(overwrite=True, drop=True)
     await async_index.delete(drop=True)
 
     with pytest.raises(RedisSearchError):
         await async_index.info()
+
+
+@pytest.mark.asyncio
+async def test_search_index_that_does_not_own_client_context_manager(async_index):
+    async with async_index:
+        await async_index.create(overwrite=True, drop=True)
+        assert async_index._redis_client
+        client = async_index._redis_client
+    assert async_index._redis_client == client
+
+
+@pytest.mark.asyncio
+async def test_search_index_that_does_not_own_client_context_manager_with_exception(
+    async_index,
+):
+    try:
+        async with async_index:
+            await async_index.create(overwrite=True, drop=True)
+            client = async_index._redis_client
+            raise ValueError("test")
+    except ValueError:
+        pass
+    assert async_index._redis_client == client
+
+
+@pytest.mark.asyncio
+async def test_search_index_that_does_not_own_client_disconnect(async_index):
+    await async_index.create(overwrite=True, drop=True)
+    client = async_index._redis_client
+    await async_index.disconnect()
+    assert async_index._redis_client == client
+
+
+@pytest.mark.asyncio
+async def test_search_index_that_does_not_own_client_disconnect_sync(async_index):
+    await async_index.create(overwrite=True, drop=True)
+    client = async_index._redis_client
+    async_index.disconnect_sync()
+    assert async_index._redis_client == client
+
+
+@pytest.mark.asyncio
+async def test_search_index_that_owns_client_context_manager(index_schema, redis_url):
+    async_index = AsyncSearchIndex(schema=index_schema, redis_url=redis_url)
+    async with async_index:
+        await async_index.create(overwrite=True, drop=True)
+        assert async_index._redis_client
+    assert async_index._redis_client is None
```

Comment:
So if providing a connection URL to let redisvl manage your connection, it is not maintaining that connection when using the context manager. That makes sense, but does also mean the usage should ideally be well documented

Reply:
So it's actually the opposite if I understood you: if you give it a client object that means you are managing the object, but if you give it connection details and it creates the client object, then it manages the object and will disconnect and null it out when the context exits.

Reply:
We're on the same page. cool